server/mysterium_api_fake.go

```diff
@@ -29,6 +29,28 @@ func (client *ClientFake) RegisterProposal(proposal dto_discovery.ServiceProposa
 	return nil
 }
 
+//UnregisterProposal unregisters a service proposal when client disconnects
+func (client *ClientFake) UnregisterProposal(providerID string, signer identity.Signer) error {
+	matchingProposals, err := client.FindProposals(providerID)
```

Comment:
I suggest not to couple on function `FindProposals`, just traverse `client.proposalsMock` and delete matched values